rad/prelude/cmd-parsing.rad

```diff
@@ -27,32 +29,57 @@
   (fn [cmd]
     (string-append "Too many arguments for command \"" cmd "\"")))
 
-(def /cmd-1-opts
-  "Matches a command `cmd` with one argument and an optional
-  combination of options that are allowed by opts-map"
-  (fn [cmd opts opts-map]
+(def /cmd-opts
+  "Matches the command `cmd` and a combination of optional parameters, where
+  `opt-match` matches the options to the keys of `multi-ops`. `multi-ops` is a
+  dict of option-keys with multiple arguments. If no options are passed, the
+  `default` is returned."
+  (fn [cmd opts multi-opts opt-match default help]
```

Comment:
What do you think about the following alternative API for `/cmd-opts`. At the moment we have an argument for every attribute of an option where the argument maps option names to attribute. Instead I’d suggest having only a list of option specifiers as an argument. For example this would look like this
~~~clojure
(/cmd-opts "list" [
    { :key :state :type :multi-option :options ["-s", "--state", "--filter-by-state"]  :possible-values ["acc", "closed"] }
    { :key :fancy :type :flag :options ["--fancy"] :default #f }
    ])
~~~